py/gc.h

```diff
@@ -48,6 +48,11 @@ void gc_collect_end(void);
 // Use this function to sweep the whole heap and run all finalisers
 void gc_sweep_all(void);
 
+#if MICROPY_GC_PERMANENT_ALLOCS
+// Mark an allocation as "permanent" (i.e. will survive soft resets / heap re-init).
```

Comment:
how about "Mark an allocation as "permanent": it will never be collected, not even by gc_sweep_all."

Reply:
That sounds more confusing to me. I have no idea when gc_sweep_all may be called... I do understand what a soft reset is...

Reply:
But the GC doesn't have a concept of soft-reset, MPY soft-reset is defined at a higher level.  And in particular it won't survive a heap re-init via `gc_init()`.